initial commit for auth and user service

xx-db opened the pull request:
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

<!-- Please state what you've changed and how it might affect the users. -->

xx-db committed:
initial commit for auth and user service

vikrantpuppala commented on server/src/main/java/io/unitycatalog/server/service/PrivilegeAssignment.java:
```diff
@@ -0,0 +1,13 @@
+
+@Getter
+@Builder
+class PrivilegeAssignment {
```
should we define this in the spec and let it be autogenerated?

creechy replied:
Ya. Not a bad idea.

vikrantpuppala commented on server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationResponse.java:
```diff
@@ -0,0 +1,11 @@
+@Getter
+@Builder
+public class UpdateAuthorizationResponse {
+  List<PrivilegeAssignment> privilege_assignments;
```
camel case

vikrantpuppala commented on server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationResponse.java:
move to openapi spec and autogenerate?

vikrantpuppala commented on server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationRequest.java:
move to openapi spec and autogenerate?

xx-db committed:
use auto generated api model

xx-db committed:
integrate with the JCasbinAuthenticator commit

xx-db committed:
integrate with jcasbin jdbc adaptor

does not work with H2, needs to switch to mysql
also fixed issue with overriding hibernate config in HibernateUtils
